File: py/102.py
'''
    二叉树的层序遍历
'''
from typing import List

class TreeNode:
    def __init__(self, val, left, right):
        self.val = val
        self.left = left
        self.right = right

class Solution:
    def levelOrder(self, root: TreeNode) -> List[List[int]]:
        res = []

        tree = [root]

        while tree:
            tmp = []
            nextTree = []

            for tr in tree:
                if tr:
                    tmp.append(tr.val)
                    nextTree.extend([tr.left, tr.right])
            if tmp:
                res.append(tmp)
            tree = nextTree

        return res

File: cpp/102.cpp
#include <iostream>
#include <vector>
using namespace std;

struct TreeNode
{
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode(int x) : val(x), left(NULL), right(NULL) {}
};

class Solution
{
public:
    vector<vector<int>> result;
    int num;
    void solve(TreeNode *root, int n)
    {
        if (root == NULL)
            return;
        if (result.size() < n + 1)
        {
            vector<int> t;
            result.push_back(t);
        }
        result[n].push_back(root->val);
        solve(root->left, n + 1);
        solve(root->right, n + 1);
        return;
    }
    vector<vector<int>> levelOrder(TreeNode *root)
    {
        solve(root, 0);
        return result;
    }
};